lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/EvaluatorManager.cs

```diff
@@ -0,0 +1,312 @@
+        internal const string MasterBatchId = "MasterBatchId";
+        internal const string MapperBatchId = "MapperBatchId";
+
+        private readonly ISet<string> _allocatedEvaluatorIds = new HashSet<string>();
+        private readonly ISet<string> _failedEvaluatorIds = new HashSet<string>();
+
+        private readonly int _totalExpectedEvaluators;
+        private readonly int _allowedNumberOfEvaluatorFailures;
+        private readonly IEvaluatorRequestor _evaluatorRequestor;
+        private string _masterEvaluatorId;
+
+        private readonly EvaluatorSpecification _updateEvaluatorSpecification;
+        private readonly EvaluatorSpecification _mapperEvaluatorSpecification;
+
+        /// <summary>
+        /// Creates a EvaluatorManager for driver which contains specification for the Evaluators
+        /// </summary>
+        /// <param name="totalEvaluators"></param>
+        /// <param name="allowedNumberOfEvaluatorFailures"></param>
+        /// <param name="evaluatorRequestor"></param>
+        /// <param name="updateEvaluatorSpecification"></param>
+        /// <param name="mapperEvaluatorSpecification"></param>
+        internal EvaluatorManager(
+            int totalEvaluators,
+            int allowedNumberOfEvaluatorFailures,
+            IEvaluatorRequestor evaluatorRequestor,
+            EvaluatorSpecification updateEvaluatorSpecification,
+            EvaluatorSpecification mapperEvaluatorSpecification)
+        {
+            _totalExpectedEvaluators = totalEvaluators;
+            _allowedNumberOfEvaluatorFailures = allowedNumberOfEvaluatorFailures;
+            _evaluatorRequestor = evaluatorRequestor;
+            _updateEvaluatorSpecification = updateEvaluatorSpecification;
+            _mapperEvaluatorSpecification = mapperEvaluatorSpecification;
+        }
+
+        /// <summary>
+        /// Request update/master Evaluator from resource manager
+        /// </summary>
+        internal void RequestUpdateEvaluator()
+        {
+            _evaluatorRequestor.Submit(
+                _evaluatorRequestor.NewBuilder()
+                    .SetCores(_updateEvaluatorSpecification.Core)
+                    .SetMegabytes(_updateEvaluatorSpecification.Megabytes)
+                    .SetNumber(1)
+                    .SetEvaluatorBatchId(MasterBatchId)
+                    .Build());
+        }
+
+        /// <summary>
+        /// Request map evaluators from resource manager
+        /// </summary>
+        /// <param name="numEvaluators">Number of evaluators to request</param>
+        internal void RequestMapEvaluators(int numEvaluators)
+        {
+            _evaluatorRequestor.Submit(
+                _evaluatorRequestor.NewBuilder()
+                    .SetMegabytes(_mapperEvaluatorSpecification.Megabytes)
+                    .SetNumber(numEvaluators)
+                    .SetCores(_mapperEvaluatorSpecification.Core)
+                    .SetEvaluatorBatchId(MapperBatchId)
```

Comment:
This is a problem here. Suppose earlier you asked for 100 mappers with batch id as MapperBatchId. Now if 5 fail and you use the function again to request for them, you cannot use the same batch Id. It will throw an error.

@afchung @markusweimer  Can you confirm this logic?